bliss/simulator/prior.py

```diff
@@ -95,18 +99,72 @@ def __init__(
         self.galaxy_a_scale = galaxy_a_scale
         self.galaxy_a_bd_ratio = galaxy_a_bd_ratio
 
-    def sample_prior(self) -> TileCatalog:
+        # load all star, galaxy ratios required for sampling
+        if self.n_bands == 1:
+            return
+        for field_params in sdss_fields["field_list"]:
```

Comment:
(L105 - 137) Constructs `self.rcf_stars_rest` and `self.rcf_gals_rest`, which contain the relative flux values across all bands for a given RCF depending on whether the source is a star or galaxy. This change ensures that all files are read in the constructor up-front as opposed to having a random file read whenever a batch is generated. Additionally, it fixes the issue with the rejection-sampling scheme from earlier. This avoids the issues with variable runtime.